src/tests/eventpipe/ThreadPoolEvents.cs

```diff
@@ -0,0 +1,168 @@
+                        Logger.logger.Log("ThreadAdjustmentResult check: " + ThreadAdjustmentResult);
+                        
+                        return ThreadStartStopResult && ThreadAdjustmentResult ? 100 : -1;
+                    };
+                };
+
+                var config = new SessionConfiguration(circularBufferSizeMB: (uint)Math.Pow(2, 10), format: EventPipeSerializationFormat.NetTrace,  providers: providers);
+
+                var ret = IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, config, _DoesTraceContainEvents);
+                Assert.Equal(100, ret);
+            }, output);
+        }
+        [Fact]
+        public async void IOThread_ProducesEvents()
+        {
+            await RemoteTestExecutorHelper.RunTestCaseAsync(() => 
+            {
+                Dictionary<string, ExpectedEventCount> _expectedEventCounts = new Dictionary<string, ExpectedEventCount>()
+                {
+                    { "Microsoft-Windows-DotNETRuntime", -1 },
+                    { "Microsoft-Windows-DotNETRuntimeRundown", -1 },
+                    { "Microsoft-DotNETCore-SampleProfiler", -1 }
+                };
+
+                var providers = new List<Provider>()
+                {
+                    new Provider("Microsoft-DotNETCore-SampleProfiler"),
+                    //GCKeyword (0x1): 0b1
+                    new Provider("Microsoft-Windows-DotNETRuntime", 0b10000_0000_0000_0000, EventLevel.Informational)
+                };
+
+                string filePath = Path.Combine(Path.GetTempPath(), "Temp.txt");
+                Action _eventGeneratingAction = () => 
+                {
+                    for(int i=0; i<50; i++)
+                    {
+                        if (i % 10 == 0)
+                            Logger.logger.Log($"Create file stream {i} times...");
+
+                        FileStream fileStream = new FileStream(filePath, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.ReadWrite, 1024, true);
+                        byte[] bytes= new byte[1024 * 1024];
+                        
+                        fileStream.BeginWrite(bytes, 0, bytes.Length, new AsyncCallback(asyncCallback), fileStream);
+                        Thread.Sleep(1000);
+                    }
+                };
+                void asyncCallback(IAsyncResult result)
+                {
+                    FileStream fileStream = (FileStream)result.AsyncState;
+                    fileStream.EndWrite(result);
+                    fileStream.Close();
+                    File.Delete(filePath);
+                } 
+
+                Func<EventPipeEventSource, Func<int>> _DoesTraceContainEvents = (source) => 
+                {
+                    int IOThreadCreationStartEvents = 0;
+                    int IOThreadCreationStopEvents = 0;
+                    source.Clr.IOThreadCreationStart += (eventData) => IOThreadCreationStartEvents += 1;
```

Comment:
I looked through the runtime source code and it looks like these events are fired inside win32threadpool.cpp and are wrapped in `#ifndef FEATURE_PAL` so I don't think you'll see these show up on non-Windows platforms.  For now, let's not include this test.

Reply:
Fixed, removed this test.